src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -47,15 +67,16 @@ public partial record SourceFileInfo(
             string Contents
         );
 
-        [JsonSerializable(typeof(MetadataEntry))]
+        [JsonSerializable(typeof(ArchiveMetadata))]
         [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
         private partial class MetadataSerializationContext : JsonSerializerContext { }
 
-        [JsonSerializable(typeof(MetadataEntry))]
-        private record MetadataEntry(
+        [JsonSerializable(typeof(ArchiveMetadata))]
+        private record ArchiveMetadata(
             int MetadataVersion,
             string EntryPoint, // Path of the entrypoint file
-            IEnumerable<SourceFileInfoEntry> SourceFiles
+            IEnumerable<SourceFileInfoEntry> SourceFiles,
+            string BicepVersion = "unknown"
```

Comment:
Can you share more details about the need a default value here?

Reply:
Backwards compat - the property may not exist, and I'm showing it to the user.

Reply:
Decided to make that clearer: 1426274cb